services/fishBatches.service.ts

```diff
@@ -114,51 +108,118 @@ export default class FishBatchesService extends moleculer.Service {
       fishStocking: number;
     }>,
   ) {
-    const promises = ctx.params.batches?.map((batch: any) => {
-      return ctx.call('fishBatches.create', {
-        ...batch,
-        fishStocking: ctx.params.fishStocking,
+    if (ctx.params.batches) {
+      const promises = ctx.params.batches?.map((batch) => {
+        return ctx.call('fishBatches.create', {
+          fishType: batch.fishType,
+          fishAge: batch.fishAge,
+          amount: batch.amount,
+          weight: batch.weight,
+          fishStocking: ctx.params.fishStocking,
+        });
       });
-    });
-    await Promise.all(promises);
+      await Promise.all(promises);
+    }
   }
 
-  //TODO: different batches validation during registration, review and when editing trough admin
   @Action({
     params: {
-      batches: 'array|optional',
-      fishStocking: 'number|optional',
+      batches: {
+        type: 'array',
+        required: false,
+        properties: {
+          id: 'number|integer|positive|optional',
+          fishType: 'number|integer|positive',
+          fishAge: 'number|integer|positive',
+          amount: 'number|integer|positive',
+          weight: 'number|optional',
+          reviewAmount: 'number|integer|positive|optional',
+          reviewWeight: 'number|optional',
+        }
```

Comment:
https://github.com/icebob/fastest-validator?tab=readme-ov-file#array

Array turi `items`. Tada jau ten yra objektas ir objektas turi properties. Yra pavyzdys pateiktam linke su array of objects validacijai